From fd3a4b41a71ecafd4bf38da51632a53924f5aac3 Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yb@renesas.com>
Date: Wed, 22 Jul 2020 11:19:53 +0700
Subject: [PATCH 251/504] ARM: dts: r8a7743: Add vcp/vpc modules

Add vcp/vpc for uvcs-kernel-module to support LTS4.4.6
In details:
These are issues when porting from 3.10 to 4.4
    Issue 1: Can not request interrupt for HW IP (VCP)
    Issue 2: Can not map HW IP register (VPC)
These is a difference in register platform device for the driver:
    - The old driver use kernel API (platform_device_register())
to register it device to kernel,
    - The new driver, device binding to driver is register by the new way
that use the device info from <target-board>.dtb file.

vpc1, vpc0xy, and vpc1xy are added as reserve only vcp0, vpc0 and prr is used.
The prr is used for getting LSI information which tell the driver
what kind of board it running

Signed-off-by: Duy Dang <duy.dang.yb@renesas.com>
---
 arch/arm/boot/dts/r8a7743.dtsi | 48 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743.dtsi b/arch/arm/boot/dts/r8a7743.dtsi
index b91ec37..30ac1e4 100644
--- a/arch/arm/boot/dts/r8a7743.dtsi
+++ b/arch/arm/boot/dts/r8a7743.dtsi
@@ -1704,6 +1704,54 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				<0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7743_CLK_VCP0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7743_CLK_VPC0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A7743_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A7743_CLK_VPC0>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A7743_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		vsp@fe928000 {
 			compatible = "renesas,vsp1";
 			reg = <0 0xfe928000 0 0x8000>;
-- 
2.7.4

